api/main_endpoints/routes/UserSearch.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+const User = require('../models/User.js');
+const {
+  checkIfTokenSent,
+  checkIfTokenValid,
+} = require('../util/token-functions');
+const {
+  OK,
+  BAD_REQUEST,
+  UNAUTHORIZED,
+  FORBIDDEN,
+} = require('../../util/constants').STATUS_CODES;
+const membershipState = require('../../util/constants').MEMBERSHIP_STATE;
+
+const ROWS_PER_PAGE = 20;
+
+// Search for all members using either first name, last name or email
+router.post('/shortcutsearchusers', async function(req, res) {
+  if (!checkIfTokenSent(req)) {
+    return res.sendStatus(FORBIDDEN);
+  } else if (!checkIfTokenValid(req, membershipState.OFFICER)) {
+    return res.sendStatus(UNAUTHORIZED);
+  }
+
+  const limit = Math.min(Number(req.body.limit) || ROWS_PER_PAGE, ROWS_PER_PAGE);
+  let maybeOr = {};
+  const query = req.body.query;
+  if (query) {
+    const parts = query.trim().split(/\s+/);
+    const first = parts[0];
+    const last = parts[parts.length - 1];
+    const nameRegex = new RegExp(`^${first}.*${last}`, 'i');
```

Comment:
what if we do a match using whatever the user typed

so if i was searching for `person one`, i could match it by typing `pone`

```suggestion
    const query = req.body.query.replace(/\s/g, '');
    const pattern = new RegExp(`.*${query}.*`);
```

another example the above would match on would be `person o`, as right now the pr's logic is expecting the entire first and last name to by typed out to do a match

Reply:
New suggestion: Instead of matching the first and last name, the backend will match the record using the input sequence:
```
const query = req.body.query.replace(/\s/g, '');
const fuzzyPattern = query.split('').join('.*');
const pattern = new RegExp(fuzzyPattern, 'i');
```
This trades off O(n) performance for maximum flexibility. Note: this approach does not ignore discrepancies in middle names.